### PR TITLE
nRF91: initial support for GNSS

### PR DESCRIPTION
**Pull request by raiden00pl**
## Summary

This PR adds initial GNSS support for nRF91. It's based on the new sensor framework and GPS upper-half driver.
After a bit of work, we can forward NMEA messages to user-space, I omitted this feature as it's useless for me.
AGPS is not supported for now and there is much more to do here.

Commits:
- sensors/gps: define the number of items in the satellite info list
- drivers/sensors/Make.defs: cosmetics
- nrf91/nrf91_modem_sock.c: fix printf warnings
- nrf91/nrf91_modem.c: add support for board-specific modem configuration
- boards/thingy91: add board specific modem configuration
- boards/nrf9160-dk: add board specific modem configuration
- nrf91/nrf91_modem.c: configure system mode
- nrf91: initial support for GNSS (GPS only for now)
- boards/nrf9160-dk: enable GNSS for modem_ns config
- Documentation: update nRF91 doc

## Impact
GNSS supported for nrf91

## Testing
nrf9610-dk

```
nsh> cu -l /dev/modem # enable modem
AT+CFUN=1
OK
~.

nsh> uorb_listener -t 1000

... after some time ...

sensor_gps_satellite:   timestamp: 157380000 (0 us ago)
sensor_gps_satellite:   number:0 svid: 24 elevation: 0 azimuth: 0 snr: 40
sensor_gps_satellite:   number:1 svid: 10 elevation: 0 azimuth: 0 snr: 44
sensor_gps_satellite:   number:2 svid: 14 elevation: 0 azimuth: 0 snr: 47
sensor_gps_satellite:   number:3 svid: 2 elevation: 0 azimuth: 0 snr: 31
sensor_gps:     timestamp: 158400000 (0 us ago) time_utc: 1700144100 latitude: 52.xxxx longitude: 18.xxxx
sensor_gps:     altitude: 144.8454 altitude_ellipsoid: 0.0000 ground_speed: 0.0199 course: 0.0000
sensor_gps:     eph: 10.1487 epv: 27.6759 hdop: 2.3040 vdop: 4.8655


```

### CODE REVIEW

File: arch/arm/src/nrf91/nrf91_modem.c [hunk 1]
```diff
@@ -122,5 +159,22 @@ int nrf91_modem_initialize(void)
       nerr("nrf_modem_init failed %d\n", ret);
     }
```
Comment (xiaoxiang781216) [inline arch/arm/src/nrf91/nrf91_modem.c:160]: should we goto errout here

File: arch/arm/src/nrf91/Kconfig [hunk 2]
```diff
@@ -742,4 +742,22 @@ config NRF91_MODEM_AT
...
+
+config NRF91_MODEM_GNSS
+	bool "Modem GNSS support"
+	depends on SENSORS_GPS
+	default y
+
+config NRF91_MODEM_PREFERENCE
+	int "Modem mode preference"
+	default 0
+	range 0 4
```
Comment (acassis) [inline arch/arm/src/nrf91/Kconfig:761]: @raiden00pl is there some documentation/explanation about these 5 preference types?

File: arch/arm/src/nrf91/nrf91_modem_gnss.c [hunk 3]
```diff
@@ -0,0 +1,734 @@
...
+
+  fix_interval = (*period_us) / 1000000;
+
+  /* Handle GNSS mode */
+
+  if (fix_interval == 1)
+    {
+      /* Continuous navigation with 1 Hz rate */
+
+      fix_interval = 1;
```
[the remark this answers is not in the document]
Comment (raiden00pl) [reply arch/arm/src/nrf91/nrf91_modem_gnss.c]: some leftovers, removed

File: arch/arm/src/nrf91/nrf91_modem.c (hunk 1 above)
Comment (raiden00pl) [reply arch/arm/src/nrf91/nrf91_modem.c:170]: fixed

File: arch/arm/src/nrf91/Kconfig (hunk 2 above)
Comment (raiden00pl) [reply arch/arm/src/nrf91/Kconfig:761]: I added help message and a note about the doc where we can find some details

**Review by acassis — APPROVED** [review]

**Pull request merged**